invertedai/large/common.py

```diff
@@ -0,0 +1,153 @@
+    agent_states: Optional[List[AgentState]] = None # A list of existing agents within the region
+    agent_attributes: Optional[List[AgentAttributes]] = None # The attributes of agents that exist within the region or that will be initialized within the region
+    recurrent_states: Optional[List[RecurrentState]] = None # Recurrent states of the agents contained within the region
+    region_type: str = 'square' # Geometric category of the region. As of now, only 'square' is supported.
+    vertexes: List[Point] # An ordered list of x-y coordinates of the region defined clockwise around the perimeter of the region
+
+    @classmethod
+    def fromlist(cls, l, agent_states = [], agent_attributes = [], recurrent_states = [], region_type = 'square'):
+        center = None
+        fov = 100
+
+        if region_type == 'square':
+            if len(l) == 2:
+                center, fov = l
+
+            elif len(l) == 1:
+                center = l[0]
+
+            vertexes = cls.define_square_vertices(cls,center,fov)
+
+        for agent in agent_states:
+            assert cls.check_point_in_region(cls,agent.center), f"Existing agent states at position {agent.center} must be located within the region."
+
+        return cls(
+            center=center, 
+            fov=fov,
+            region_type=region_type, 
+            vertexes=vertexes,
+            agent_states=agent_states, 
+            agent_attributes=agent_attributes, 
+            recurrent_states=recurrent_states 
+        )
+
+    def clear_agents(self):
+        self.agent_states = None
+        self.agent_attributes = None
+        self.recurrent_states = None
+
+    def insert_all_agent_details(self,agent_state,agent_attributes,recurrent_state):
+        if self.agent_states is None:
+            self.agent_states = [agent_state]
+        else:
+            self.agent_states.append(agent_state)
+
+        if self.agent_attributes is None:
+            self.agent_attributes = [agent_attributes]
+        else:
+            self.agent_attributes.append(agent_attributes)
+
+        if self.recurrent_states is None:
+            self.recurrent_states = [recurrent_state]
+        else:
+            self.recurrent_states.append(recurrent_state)
+
+    def define_square_vertices(self,center,fov):
+        assert center is not None, f"Square region must contain valid center Point."
+        fov_split = fov/2
+        center_x = center.x
+        center_y = center.y
+        vertexes = [
+            Point.fromlist([center_x-fov_split,center_y+fov_split]), # Top left
+            Point.fromlist([center_x+fov_split,center_y+fov_split]), # Top right
+            Point.fromlist([center_x+fov_split,center_y-fov_split]), # Bottom right
+            Point.fromlist([center_x-fov_split,center_y-fov_split]) # Bottom left
+        ]
+
+        return vertexes
+
+    def check_point_in_region(self,point):
```

Comment:
It makes sense to have a doc string for this method explaining how things work in general, but you should reformat it  to have it enclosed instead of using the comments. 
And i don't think you need those additional comments explain how things work line by line